src/mmrelay/matrix_utils.py

```
@@ -1287,55 +1289,72 @@ async def connect_matrix(
+            return None
 
-    # If credentials.json exists, use it
-    if credentials:
-        matrix_homeserver = credentials["homeserver"]
-        matrix_access_token = credentials["access_token"]
-        bot_user_id = credentials["user_id"]
-        e2ee_device_id = _get_valid_device_id(credentials.get("device_id"))
+    # Load credentials using the shared helper (supports legacy search + env overrides)
+    try:
+        credentials = await async_load_credentials()
+    except Exception as exc:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)